src/toil/provisioners/clusterScaler.py

```diff
@@ -235,6 +235,30 @@ def __init__(self, provisioner, jobBatcher, config):
         else:
             self.scaler = None
 
+    def check(self):
+        """
+        Attempt to join any existing scaler threads with a short timeout. This insures an exception
+        raised in the threads is propagated in a timely fashion. If there are exceptions in both threads,
+        one exception is logged before the other is raised to insure that both tracebacks are visible
+        """
+        exceptions = []
+        if self.preemptableScaler:
+            try:
+                self.preemptableScaler.join(1)
+            except Exception as e:
+                exceptions.append(e)
+        if self.scaler:
+            try:
+                self.scaler.join(1)
+            except Exception as e:
+                exceptions.append(e)
+        if len(exceptions) > 1:
+            # we want stacktraces for both so we have to log one exception before raising the other
```

Comment:
Addressing my comment above would make the list and this special casing obsolete.

I'd be ok with raising a secondary exception if the primary ones are being logged with stack trace.

Independently, I'd feel strange about the asymmetric handling of those two exceptions. But probably unfounded.